### ISSUE TITLE
About reproducing the result of COCOA

### ISSUE THREAD

**Issue by MaginaDai** [cruiseresearchgroup/COCOA#1]
Hi authors, 
  I tried to reproduce the experiment results in your paper based on your code, but I encountered some problems. Could you please help me out? Questions are listed below:
  1. Which loss_type should I choose, 'cocoa2' or 'cocoa'? I think both are not aligned with the loss function described in the paper. 'cocoa' is in an InforNCE-like form but the loss function of the paper adds positive error and negative error up. 'cocoa2' adds them up. However, the positive error considers the similarity of the same sensor (do not mask the similarity on the diagonal positions) (Line 91-96 in losses.py). As for the negative error, Equation (3) in the paper states that similarity refers to the similarity across different timestamps of the same modality. However, the sim in Line 100 in losses.py contains the similarity across different timestamps of the different modalities. Could you please clarify it?
  2. How to select the hyperparameters lambd and scale_loss for CustomLoss in losses.py? In your code lambd = 3.9e-3, which makes the negative error quite small. Then it contributes little to the final loss. Based on my observation, the positive error (more than thousands) is usually much larger than the negative error (as small as one or two). May I ask what's the hyperparameters to reproduce your experiment results? 

Looking forward to your reply.

Best
Gaole Dai
SCSE, NTU, Singapore
